src/bsp/Bsp.cpp

```diff
@@ -2802,6 +2860,19 @@ int Bsp::add_texture(const char* name, byte* data, int width, int height) {
 		}
 	}
 
+	if (oldtex != nullptr)
+	{
+		memcpy((byte*)oldtex + oldtex->nOffsets[0], mip[0], width * height);
+		memcpy((byte*)oldtex + oldtex->nOffsets[1], mip[1], (width >> 1) * (height >> 1));
+		memcpy((byte*)oldtex + oldtex->nOffsets[2], mip[2], (width >> 2) * (height >> 2));
+		memcpy((byte*)oldtex + oldtex->nOffsets[3], mip[3], (width >> 3) * (height >> 3));
+		memcpy((byte*)oldtex + (oldtex->nOffsets[3] + (width >> 3) * (height >> 3) + 2), palette, sizeof(COLOR3) * 256);
+		for (int i = 0; i < MIPLEVELS; i++) {
+			delete[] mip[i];
+		}
+		return 0;
+	}

```

Comment:
the new texture might be a different size, which could cause a crash or corruption.

Reply:
Yes I can rename new texture to another  if w/h different

Reply:
Would make more sense to replace the existing one, but this code is kind of a pain to work with so renaming is fine too.

Reply:
Ok

Reply:
still a problem with the latest code

Reply:
Sorry i skip your message)